src/company_cmd.cpp

```diff
@@ -1179,3 +1180,54 @@ uint32 CompanyInfrastructure::GetTramTotal() const
 	}
 	return total;
 }
+
+void CcGiveMoney(const CommandCost &result, TileIndex tile, uint32 p1, uint32 p2, uint32 cmd)
+{
+	if (result.Failed() || !_settings_game.economy.give_money) return;
+
+	/* Inform the company of the action of one of its clients (controllers). */
+	char msg[64];
+	SetDParam(0, p2);
+	GetString(msg, STR_COMPANY_NAME, lastof(msg));
+
+	if (!_network_server) {
+		NetworkClientSendChat(NETWORK_ACTION_GIVE_MONEY, DESTTYPE_TEAM, p2, msg, p1);
+	} else {
+		NetworkServerSendChat(NETWORK_ACTION_GIVE_MONEY, DESTTYPE_TEAM, p2, msg, CLIENT_ID_SERVER, p1);
+	}
+}
+
+/**
+ * Transfer funds (money) from one company to another.
+ * To prevent abuse in multiplayer games you can only send money to other
+ * companies if you have paid off your loan (either explicitly, or implicitly
+ * given the fact that you have more money than loan).
+ * @param tile unused
+ * @param flags operation to perform
+ * @param p1 the amount of money to transfer; max 20.000.000
+ * @param p2 the company to transfer the money to
+ * @param text unused
+ * @return the cost of this operation or an error
+ */
+CommandCost CmdGiveMoney(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32 p2, const char *text)
+{
+	if (!_settings_game.economy.give_money) return CMD_ERROR;
+
+	const Company *c = Company::Get(_current_company);
+	CommandCost amount(EXPENSES_OTHER, min((Money)p1, (Money)20000000LL));
+	CompanyID dest_company = (CompanyID)p2;
+
+	/* You can only transfer funds that is in excess of your loan */
+	if (c->money - c->current_loan < amount.GetCost() || amount.GetCost() < 0) return CMD_ERROR;
+	if (!_networking || !Company::IsValidID(dest_company)) return CMD_ERROR;
```

Comment:
Possibly also not in scope of this PR:
* Either allow transfering money in singleplayer.
* Or do not allow transfering money to AIs in multiplayer.

Reply:
I see no problem allowing money transfers to AIs tbh :) But indeed, in singleplayer this should now also "just" work :)